### PR TITLE
Implementation of the findings framework

### PR DESCRIPTION
**Pull request by ericatcorsha**
Before you submit your PR, make sure you check each of the following:

1. [x] I have read the [Contributing Guidelines](./CONTRIBUTING.md) and signed the [Contributor License Agreement](./CLA.md).
2. [x] My code is clean and ready-to-PR, including being free of lint errors and dead code
3. [x] I have **tested** my change / code and am confident that it works
4. [x] I have filled in the template below completely and accurately
##### Changelog

- Implemented the analysis framework
- Ported the reused authentication analysis to the framework
- Implemented the `/api/analyses` endpoint that runs all analyses.

### CODE REVIEW

File: ui/src/lib/findings.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+/** When a finding occurred. It might be a time span or at an instant */
+export type OccurredAt =
+  | {
+      /** The start date and time formatted as a UTC RFC-3339 string */
+      start: string;
+
+      /** The end date and time formatted as a UTC RFC-3339 string */
+      end;
+      string;
```
Comment (ArcticLight) [inline ui/src/lib/findings.ts:49]: should probably be like this (?)
```suggestion
      /** The end date and time formatted as a UTC RFC-3339 string */
      end: string;
```

File: ui/src/lib/findings.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+   * A raw string field that can be used to display additional
+   * free-form information in the UI about the finding in
+   * CommonMark */
+  detail: string;
+}
+
+/** 
+    Evaluates and sort all analyses in ascending order based on priority
+*/
+export type AnalysisFunction = () => Promise<Analysis>;
```
Comment (ArcticLight) [inline ui/src/lib/findings.ts:100]: Doc comment doesn't describe exported type - this is the comment for the function below, perhaps?

**Review by ArcticLight — CHANGES_REQUESTED** [review]
Just two nits and we can get the definition here finalized

File: ui/src/lib/findings.ts (hunk 2 above)
Comment (ericatcorsha) [reply ui/src/lib/findings.ts]: fixed

File: ui/src/lib/findings.ts (hunk 1 above)
Comment (ericatcorsha) [reply ui/src/lib/findings.ts]: fixed

File: ui/src/lib/findings.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
+/* Copyright 2022 Corsha.
```
Comment (DavidCorsha) [inline ui/src/lib/findings.ts:1]: 2023 please

**Review by DavidCorsha — CHANGES_REQUESTED** [review]

File: ui/src/lib/findings.ts (hunk 3 above)
Comment (ericatcorsha) [reply ui/src/lib/findings.ts]: done

**Review by DavidCorsha — APPROVED** [review]

**Review by ArcticLight — APPROVED** [review]

**Review by DavidCorsha — DISMISSED** [review]

File: k8s/helm/cast/files/init.sql [hunk 4]
```diff
@@ -12,7 +12,10 @@
 -- DROP TABLE IF EXISTS traffic;
 CREATE TABLE IF NOT EXISTS traffic (
   id uuid PRIMARY KEY DEFAULT gen_random_uuid(),
+  occurred_at timestamptz,
   data jsonb
 );
 
 CREATE INDEX IF NOT EXISTS idx_traffic_data ON traffic USING gin (data);
+CREATE INDEX IF NOT EXISTS idx_auth_header ON traffic USING BTREE (data->'request'->'headers'->>'Authorization');
+CREATE INDEX IF NOT EXISTS idx_auth_header_src ON traffic USING BTREE (data->'request'->'headers'->>'Authorization', data->'src');
```
Comment (DavidCorsha) [inline k8s/helm/cast/files/init.sql:21]: Should this be part of a separate PR?

**Review by DavidCorsha — CHANGES_REQUESTED** [review]

File: k8s/helm/cast/files/init.sql (hunk 4 above)
Comment (ericatcorsha) [reply k8s/helm/cast/files/init.sql:21]: no, it is part of the reused authorization analysis. The index prevents a sequential scan of the table.

**Review by DavidCorsha — APPROVED** [review]

**Review by ArcticLight — APPROVED** [review]

**Pull request merged**